dev/deps/spark-deps-hadoop-3-hive-2.3

```diff
@@ -64,17 +65,18 @@ gcs-connector/hadoop3-2.2.7/shaded/gcs-connector-hadoop3-2.2.7-shaded.jar
 gmetric4j/1.0.10//gmetric4j-1.0.10.jar
 gson/2.2.4//gson-2.2.4.jar
 guava/14.0.1//guava-14.0.1.jar
-hadoop-aliyun/3.3.4//hadoop-aliyun-3.3.4.jar
-hadoop-annotations/3.3.4//hadoop-annotations-3.3.4.jar
-hadoop-aws/3.3.4//hadoop-aws-3.3.4.jar
-hadoop-azure-datalake/3.3.4//hadoop-azure-datalake-3.3.4.jar
-hadoop-azure/3.3.4//hadoop-azure-3.3.4.jar
-hadoop-client-api/3.3.4//hadoop-client-api-3.3.4.jar
-hadoop-client-runtime/3.3.4//hadoop-client-runtime-3.3.4.jar
-hadoop-cloud-storage/3.3.4//hadoop-cloud-storage-3.3.4.jar
-hadoop-openstack/3.3.4//hadoop-openstack-3.3.4.jar
+hadoop-aliyun/3.3.5//hadoop-aliyun-3.3.5.jar
+hadoop-annotations/3.3.5//hadoop-annotations-3.3.5.jar
+hadoop-aws/3.3.5//hadoop-aws-3.3.5.jar
+hadoop-azure-datalake/3.3.5//hadoop-azure-datalake-3.3.5.jar
+hadoop-azure/3.3.5//hadoop-azure-3.3.5.jar
+hadoop-client-api/3.3.5//hadoop-client-api-3.3.5.jar
+hadoop-client-runtime/3.3.5//hadoop-client-runtime-3.3.5.jar
+hadoop-cloud-storage/3.3.5//hadoop-cloud-storage-3.3.5.jar
+hadoop-cos/3.3.5//hadoop-cos-3.3.5.jar
```

Comment:
Ditto. Is this added back?

Reply:
mmm. provided it doesn't interfere with everyone else, then getting it means spark will work out the box with that storage.

Reply:
I'm just worry about `HADOOP-18307 Remove hadoop-cos as a dependency of hadoop-cloud-storage` situation.

Reply:
@dongjoon-hyun Do we need to more additional check for this dependency?

Reply:
I'm just curious whether similar issue as described in https://issues.apache.org/jira/browse/HADOOP-18159 could happen again if we include `hadoop-cos` and `cos_api-bundle` in Spark's class path. We actually just ran into this exact issue recently :)

It'd be nice if there is an easy way to make this optional.